jetbrains-core/tst/software/aws/toolkits/jetbrains/services/codewhisperer/CodeWhispererLanguageManagerTest.kt

```diff
@@ -3,153 +3,173 @@
 import com.intellij.testFramework.DisposableRule
-import com.intellij.testFramework.replaceService
 import org.assertj.core.api.Assertions.assertThat
 import org.junit.Rule
 import org.junit.Test
-import org.mockito.internal.verification.Times
 import org.mockito.kotlin.doReturn
-import org.mockito.kotlin.eq
 import org.mockito.kotlin.mock
-import org.mockito.kotlin.spy
-import org.mockito.kotlin.verify
 import software.aws.toolkits.jetbrains.services.codewhisperer.language.CodeWhispererLanguageManager
-import software.aws.toolkits.jetbrains.services.codewhisperer.language.toCodeWhispererLanguage
-import software.aws.toolkits.jetbrains.services.codewhisperer.language.toProgrammingLanguage
-import software.aws.toolkits.jetbrains.services.codewhisperer.model.CaretContext
-import software.aws.toolkits.jetbrains.services.codewhisperer.model.FileContextInfo
-import software.aws.toolkits.jetbrains.services.codewhisperer.model.ProgrammingLanguage
-import software.aws.toolkits.jetbrains.services.codewhisperer.service.CodeWhispererService
+import software.aws.toolkits.jetbrains.services.codewhisperer.language.CodeWhispererProgrammingLanguage
+import software.aws.toolkits.jetbrains.services.codewhisperer.language.languages.CodeWhispererJava
+import software.aws.toolkits.jetbrains.services.codewhisperer.language.languages.CodeWhispererJavaScript
+import software.aws.toolkits.jetbrains.services.codewhisperer.language.languages.CodeWhispererJsx
+import software.aws.toolkits.jetbrains.services.codewhisperer.language.languages.CodeWhispererPlainText
+import software.aws.toolkits.jetbrains.services.codewhisperer.language.languages.CodeWhispererPython
+import software.aws.toolkits.jetbrains.services.codewhisperer.language.languages.CodeWhispererUnknownLanguage
 import software.aws.toolkits.jetbrains.utils.rules.PythonCodeInsightTestFixtureRule
 import software.aws.toolkits.telemetry.CodewhispererLanguage
+import kotlin.reflect.full.createInstance
+import kotlin.reflect.full.primaryConstructor
+import kotlin.reflect.jvm.isAccessible
 
 class CodeWhispererLanguageManagerTest {
     @Rule
     @JvmField
-    var projectRule = PythonCodeInsightTestFixtureRule()
+    val applicationRule = ApplicationRule()
 
     @Rule
     @JvmField
-    val disposableRule = DisposableRule()
+    val projectRule = PythonCodeInsightTestFixtureRule()
 
-    @Test
-    fun `test ProgrammingLanguage class`() {
-        // isEqualTo() is based on languageName field, which is case-insensitive
-        assertThat(ProgrammingLanguage("JAVA")).isEqualTo(ProgrammingLanguage("Java"))
-        assertThat(ProgrammingLanguage("JAVA").languageName).isEqualTo("java")
+    @Rule
+    @JvmField
+    val disposableRule = DisposableRule()
 
-        assertThat(ProgrammingLanguage("JavaScript")).isEqualTo(ProgrammingLanguage("javascript"))
-        assertThat(ProgrammingLanguage("JavaScript").languageName).isEqualTo("javascript")
-    }
+    val manager = CodeWhispererLanguageManager()
```

Comment:
Should this be CodeWhispererLanguageManager.getInstance()?

Reply:
sounds good